src/network/networkctl.c

```diff
@@ -1055,6 +1056,61 @@ static int link_lldp_status(int argc, char *argv[], void *userdata) {
         return 0;
 }
 
+static int link_delete_send_message(sd_netlink *rtnl, int index) {
+        _cleanup_(sd_netlink_message_unrefp) sd_netlink_message *req = NULL;
+        int r;
+
+        assert(rtnl);
+
+        r = sd_rtnl_message_new_link(rtnl, &req, RTM_DELLINK, index);
+        if (r < 0)
+                return rtnl_log_create_error(r);
+
+        r = sd_netlink_call(rtnl, req, 0, NULL);
+        if (r < 0)
+                return r;
+
+        return 0;
+}
+
+static int link_delete(int argc, char *argv[], void *userdata) {
+        _cleanup_(sd_netlink_unrefp) sd_netlink *rtnl = NULL;
+        _cleanup_set_free_ Set *indexes = NULL;
+        char ifname[IFNAMSIZ] = "";
+        int index, r, i;
+        Iterator j;
+
+        r = sd_netlink_open(&rtnl);
+        if (r < 0)
+                return log_error_errno(r, "Failed to connect to netlink: %m");
+
+        indexes = set_new(NULL);
+        if (!indexes)
+                return log_oom();
+
+        for (i = 1; i < argc; i++) {
+                index = (int) if_nametoindex(argv[i]);
```

Comment:
Maybe, it is better to try `parse_ifindex()` at first.
Then we can call `networkctl delete 4 veth99 dummy99 13` or something.

Reply:
That would be inconvenient for the user.

Reply:
I mean, that user can specify either ifindex or ifname.

Reply:
yeah good idea. Added now. Thanks.